src/server/dcp_server/utils.py

```diff
@@ -72,9 +76,12 @@ def crop_centered_padded_patch(x: np.ndarray,
         x[m] = 0
         if noise_intensity is not None:
             x[m] = np.random.normal(scale=noise_intensity, size=x[m].shape)
+            if mask is not None:
```

Comment:
remove? you already check if mask is not None in line 71, so this will always be true